src/arch/helperpower_128.h

```diff
@@ -53,29 +72,42 @@ static INLINE int vavailability_i(int name) { return 3; }
 
 static INLINE void vprefetch_v_p(const void *ptr) { }
 
-static vint2 vloadu_vi2_p(int32_t *p) { return vec_ld(0, p); }
-static void vstoreu_v_p_vi2(int32_t *p, vint2 v) { vec_st(v, 0, p); }
-static vint vloadu_vi_p(int32_t *p) { return vec_ld(0, p); }
-static void vstoreu_v_p_vi(int32_t *p, vint v) { vec_st(v, 0, p); }
-
-static INLINE vdouble vload_vd_p(const double *ptr) { return (vector double)vec_ld(0, (const int *)ptr); }
-static INLINE void vstore_v_p_vd(double *ptr, vdouble v) { vec_st((vector int)v, 0, (int *)ptr); }
-static INLINE vdouble vloadu_vd_p(const double *ptr) { return (vector double) ( ptr[0], ptr[1] ); }
-static INLINE void vstoreu_v_p_vd(double *ptr, vdouble v) { ptr[0] = v[0]; ptr[1] = v[1]; }
+static vint2 vloadu_vi2_p(int32_t *p)
+{ return vec_vsx_ld(0, p); }
```

Comment:
@edelsohn, @shibatch, Casting can be used for loading from memory, e.g. `*((vec_double2*)p)`, it doesn't work on CLANG and it seems works on GCC.

Reply:
Vector assignment should work with Clang.  Can you provide a specific example of an idiom that doesn't work? And by doesn't work do you mean syntax error, compiler error or wrong code?  If one lies about alignment with type punning it can produce incorrect code with any compiler.

Reply:
sorry for the delayed response, Load/Store via (cast & dereference) is buggy with almost all clang versions(except clang-4) and gcc8(older versions are working just fine) when VSX3(power9/ISA3.00) isn't enabled, these compilers tend to generate instructions 'lvx/stvx' instead of 'lxvd2x/lxvw4x/stxvd2x/stxvw4x', which lead to break the accuracy tests when unaligned memory addresses are involved.
please check the full report at https://github.com/seiko2plus/vsx_mem_test.

However, I update it the patch to use load/store via (cast & dereference) only when VSX3 is enabled or with GCC versions up to 7 when VSX(2.07) is enabled